assets/js/components/ProjectEditorTextField.js

```diff
@@ -0,0 +1,184 @@
+import $ from 'jquery'
+import { MDCTextField } from '@material/textfield'
+import { CustomTranslationApi } from '../api/CustomTranslationApi'
+
+export function ProjectEditorTextField (projectDescriptionCredits, programId, programSection, hasDefaultText, snackbar) {
+  const self = this
+  this.programId = programId
+  this.programSection = programSection
+  this.hasDefaultText = hasDefaultText
+  this.snackbar = snackbar
+  this.customTranslationApi = new CustomTranslationApi(programSection)
+
+  this.editText = $('#edit-' + this.programSection + '-text')
+  this.editTextError = $('#edit-' + this.programSection + '-text-error')
+  this.textLoadingSpinner = $('#edit-' + this.programSection + '-loading-spinner')
+  this.selectedLanguage = $('#edit-selected-language')
+
+  this.defaultSection = $('#' + this.programSection)
+  this.initialText = this.defaultSection.text().trim()
+
+  new MDCTextField(document.querySelector('#edit-' + this.programSection + '-mdc-text-field'))
+
+  this.translationSaved = projectDescriptionCredits.data('trans-translation-saved')
+  this.translationDeleted = projectDescriptionCredits.data('trans-translation-deleted')
+
+  this.pathEditName = projectDescriptionCredits.data('path-edit-program-name')
+  this.pathEditDescription = projectDescriptionCredits.data('path-edit-program-description')
+  this.pathEditCredits = projectDescriptionCredits.data('path-edit-program-credits')
+
+  this.setText = () => {
+    this.editText.val(this.initialText)
+    this.lastSavedText = this.initialText
+  }
+
+  this.hideError = () => {
+    self.editTextError.hide()
+  }
+
+  this.areChangesSaved = () => {
+    return this.editText.val().trim() === this.lastSavedText.trim()
+  }
+
+  this.isDisabled = (languageSelected) => {
+    return !this.hasDefaultText && !(languageSelected === '' || languageSelected === 'default')
+  }
+
+  this.save = (languageSelected) => {
+    const newText = this.editText.val().trim()
+    if (this.areChangesSaved() || this.isDisabled(languageSelected)) {
```

Comment:
I'm still quite confused about why we should `hideError` upon this specific condition. Can't we just `hideError` as a first thing in the `save` functions? Isn't that simpler and still work?

Reply:
Moved hideError to the beginning of this function